Add requirement that secret inputs be negotiated. (#5)
jhoyla/draft-jhoyla-tls-extended-key-schedule: draft-jhoyla-tls-extended-key-schedule.md

This structure mirrors the Handshake Injection point.

# Key Schedule Injection Negotiation

> Applications which make use of additional key schedule inputs MUST define an
> extension that negotiates the content and type of that input.  This input MUST

Reviewer: ```suggestion
Applications which make use of additional key schedule inputs MUST define a
mechanism for negotiating the content and type of that input.  This input MUST
```

be framed in a KeyScheduleSecret struct, as defined in {{structure}}.
Applications must take care that the negotiation of the content of the secret

> inputs unambiguously defines a single input.

Reviewer: I didn't parse this -- does this mean that applications can only negotiate one secret input at a time?

Author: No, this was terribly phrased. I meant that any given negotiation must have an unambiguous conclusion. It must be impossible even under adversarial conditions for the client and server to see the same extensions but believe they have negotiated different keys.

Reviewer: Perhaps we can rephrase it as such?
